src/main/java/org/terasology/climateConditions/BodyTemperatureSystem.java

```diff
@@ -0,0 +1,143 @@
+                    BodyTemperatureComponent.class, LocationComponent.class)) {
+                LocationComponent location = entity.getComponent(LocationComponent.class);
+                BodyTemperatureComponent bodyTemperature = entity.getComponent(BodyTemperatureComponent.class);
+                float envTemperature = climateConditionsSystem.getTemperature(location.getLocalPosition());
+                float envHumidity = climateConditionsSystem.getHumidity(location.getLocalPosition());
+                float deltaTemp =
+                        ((((envTemperature - (envHumidity / 10)) - bodyTemperature.current) / 200000) * CHECK_INTERVAL);
+
+                //Send event for other systems to modify change in body temperature.
+                AffectBodyTemperatureEvent affectBodyTemperatureEvent = new AffectBodyTemperatureEvent(deltaTemp);
+                entity.send(affectBodyTemperatureEvent);
+                deltaTemp = affectBodyTemperatureEvent.getResultValue();
+
+                //Check for change in body temperature levels.
+                float oldValue = bodyTemperature.current;
+                //Update current body temperature.
+                bodyTemperature.current = bodyTemperature.current + deltaTemp;
+                float newValue = bodyTemperature.current;
+                entity.saveComponent(bodyTemperature);
+                if (oldValue != newValue) {
+                    entity.send(new BodyTemperatureValueChangedEvent(oldValue, newValue));
+                }
+
+                //only for development purposes
+                entity.getOwner().send(new ChatMessageEvent("Body Temperature: " + bodyTemperature.current,
+                        entity.getOwner()));
+                entity.getOwner().send(new ChatMessageEvent("Env Temperature: " + envTemperature, entity.getOwner()));
+            }
+        }
+    }
+
+    @ReceiveEvent
+    public void onBodyTemperatureValueChanged(BodyTemperatureValueChangedEvent event, EntityRef player) {
+        BodyTemperatureLevel before = checkLevel(event.getOldValue());
+        BodyTemperatureLevel after = checkLevel(event.getNewValue());
+        if (before != after) {
+            player.send(new BodyTemperatureLevelChangedEvent(before, after));
+        }
+    }
+
+    @ReceiveEvent
+    public void onBodyTemperatureLevelChanged(BodyTemperatureLevelChangedEvent event, EntityRef player) {
+        //TODO: After Introduction of Hypo/Hyperthermia Levels, change the level of thermia corresponding to each
+        // temperature level
+        switch (event.getNewValue()) {
+            case CRITICAL_LOW:
+            case LOW:
+            case REDUCED:
+                player.addOrSaveComponent(new HypothermiaComponent());
+                break;
+            case RAISED:
+            case HIGH:
+            case CRITICAL_HIGH:
+                player.addOrSaveComponent(new HyperthermiaComponent());
```

Comment:
Should this be in level 3 already when added in critical high state?
```suggestion
                player.addOrSaveComponent(new HyperthermiaComponent(3));
```